src/components/SyncView.js

```diff
@@ -1,75 +1,140 @@
+}
+
+
+var SyncButton = styled.button`
+  background-color: orange;
+  padding: 0px 20px;
+`
+
+var Subtitle = styled.div`
+  background-color: var(--main-bg-color);
+  color: white;
+  vertical-align: middle;
+  padding: 5px 15px;
+`
+
+var LoadingText = styled.div`
+  background-color: white;
+  color: grey;
+  text-align: center;
+  min-height: 300px;
+  font-style: italic;
+  font-size: 24px;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-around;
+`
+
+var Target = styled.li`
+  min-width: 500px;
+  padding: 20px;
+  border-bottom: 1px solid grey;
+  display: flex;
+  justify-content: space-between;
+  line-height: 30px;
+  .target {
+    vertical-align: middle;
+    font-weight: bold;
+    font-size: 16px;
+  }
+  .info {
+    padding-left: 10px;
+    font-weight: normal;
+    font-size: 14px;
+    font-style: italic;
+  }
 }
+`
+
+var TargetsDiv = styled.div`
+  min-height: 300px;
+  background-color: white;
+  color: black;
+`
 
 export default class SyncView extends React.Component {
   constructor (props) {
     super(props)
-    var self = this
-    var ready = 'replication-ready'
     this.state = {
-      message: messages[ready],
-      status: ready
+      targets: [],
+      wifis: {},
+      files: {}
     }
+    this.streams = {}
+  }
 
-    ipcRenderer.on('select-file', function (event, file) {
-      if (!file) return
-      self.selectFile(file)
+  replicate (target) {
+    var self = this
+    if (!target) return
+    var stream = replicate.start(target)
+    var id = randombytes(16).toString('hex')
+    this.streams[id] = stream
+    stream.on('data', function (data) {
+      var row = JSON.parse(data)
+      var status = row.topic
+      var message = messages[status] || row.message
+      // TODO: this is clunky, improve status rendering via external module?
+      var msg = { status, message, target }
+      if (target.host) self.state.wifis[target.host] = msg
+      if (target.filename) self.state.files[target.filename] = msg
+      self.setState({wifis: self.state.wifis, files: self.state.files})
     })
-  }
 
-  onClose () {
-    this.props.changeView(MapEditor)
-  }
+    stream.on('error', function (err) {
+      if (err) console.error(err)
+    })
 
-  selectFile (filename) {
-    var self = this
-    replicate.start(filename, function (err, res, body) {
-      if (self.destroyed) return
-      if (err) {
-        self.setState({
-          message: 'Error: ' + err.message,
-          status: 'replication-error'
-        })
-      } else self.setState({status: 'replication-progress'})
+    stream.on('end', function () {
+      delete self.streams[id]
     })
   }
 
   componentWillUnmount () {
-    this.destroyed = true
-    this.stream.destroy()
+    var self = this
+    Object.keys(this.streams).map((k) => self.streams[k].destroy())
+    this.streams = {}
+    clearInterval(this.interval)
+    ipcRenderer.removeListener('select-file', this.selectFile.bind(this))
```

Comment:
Will this work? I think `bind` produces a brand new function that will fail comparisons. e.g.

```js
> console.log.bind(console) == console.log.bind(console)
false
```

Reply:
Yeah not sure, seems to be the recommended way I've always seen listeners removed in front-end

Reply:
I think if I use bind beforehand and for both adding and removing the listener, then it works! will fix! good catch